### PR TITLE
Add followers and relays to profiles

### LINKED ISSUE #391: Profile: add number of followers and how I'm connected

### PR DESCRIPTION
**Pull request by martindsq**
Closes #391, #465, #466 and #457

### CODE REVIEW

**Review by mplorentz — COMMENTED** [review]
This is looking pretty good. I noticed that the profile screen was slow to fetch metadata and the FollowCards often didn't at all, so I opened #496 to improve that.

Also the "Followed by" number is almost always 0 or 1 on my test account, which feels pretty broken. I think we should try adding a Filter specifically to calculate this number: we can request all kind 3 events that reference the given user when we load the profile screen. We should test it on jack and see how much data it downloads. If it's too much then I think we should add a button to calculate the number like Damus does.

File: Nos/Extensions/Localization+Nos.swift [hunk 1]
```diff
@@ -32,6 +32,10 @@ extension Localizable {
         }
     }
 
+    func localizedText(_ arguments: [String: String]) -> LocalizedStringKey {
```
[the remark this answers is not in the document]
Comment (martindsq) [reply Nos/Extensions/Localization+Nos.swift:35]: It returns a LocalizedStringKey, it is meant to be used when we want markdown in Text views using a String

File: Nos/Service/CurrentUser.swift [hunk 2]
```diff
@@ -293,6 +293,16 @@ class CurrentUser: NSObject, ObservableObject, NSFetchedResultsControllerDelegat
         let followKeys = following.keys
         return followKeys.contains(key)
     }
+
+    @MainActor func isBeingFollowedBy(author profile: Author) -> Bool {
```
[the remark this answers is not in the document]
Comment (martindsq) [reply Nos/Service/CurrentUser.swift:297]: I did that function in the same way isFollowing, just above, was made:
`@MainActor func isFollowing(author profile: Author) -> Bool`
do you think we should change both?

File: Nos/Extensions/Localization+Nos.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (mplorentz) [reply Nos/Extensions/Localization+Nos.swift:35]: Would it make sense to call it `localizedMarkdown()` or similar? If not maybe you could add a doc comment.

File: Nos/Service/CurrentUser.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (mplorentz) [reply Nos/Service/CurrentUser.swift:297]: yeah

[the remark this answers is not in the document]
Comment (martindsq) [reply Nos/Service/CurrentUser.swift:297]: Created ticket #506

**Review by mplorentz — APPROVED** [review]
I would still like to see the improvements I mentioned above to the follower count, but since this has already been deployed to the public I propose that we merge this PR but leave #391 open until the follower count works more intuitively.

**Pull request merged**